src/server/hotspot_partition_calculator.cpp

```diff
@@ -34,67 +35,84 @@ DSN_DEFINE_int64("pegasus.collector",
+    _partitions_stat_histories.emplace_back(temp);
 }
 
 void hotspot_partition_calculator::init_perf_counter(int partition_count)
 {
-    std::string counter_name;
-    std::string counter_desc;
+    std::string read_counter_name, write_counter_name;
+    std::string read_counter_desc, write_counter_desc;
     for (int i = 0; i < partition_count; i++) {
-        string partition_desc = _app_name + '.' + std::to_string(i);
-        counter_name = fmt::format("app.stat.hotspots@{}", partition_desc);
-        counter_desc = fmt::format("statistic the hotspots of app {}", partition_desc);
-        _hot_points[i].init_app_counter(
-            "app.pegasus", counter_name.c_str(), COUNTER_TYPE_NUMBER, counter_desc.c_str());
+        string read_partition_desc = _app_name + '.' + "read." + std::to_string(i);
+        read_counter_name = fmt::format("app.stat.hotspots@{}", read_partition_desc);
+        read_counter_desc = fmt::format("statistic the hotspots of app {}", read_partition_desc);
+        _hot_points[i].emplace_back(std::make_unique<dsn::perf_counter_wrapper>());
+        _hot_points[i][READ_HOTSPOT_DATA]->init_app_counter("app.pegasus",
+                                                            read_counter_name.c_str(),
+                                                            COUNTER_TYPE_NUMBER,
+                                                            read_counter_desc.c_str());
+        string write_partition_desc = _app_name + '.' + "write." + std::to_string(i);
+        write_counter_name = fmt::format("app.stat.hotspots@{}", write_partition_desc);
+        write_counter_desc = fmt::format("statistic the hotspots of app {}", write_partition_desc);
+        _hot_points[i].emplace_back(std::make_unique<dsn::perf_counter_wrapper>());
+        _hot_points[i][WRITE_HOTSPOT_DATA]->init_app_counter("app.pegasus",
+                                                             write_counter_name.c_str(),
+                                                             COUNTER_TYPE_NUMBER,
+                                                             write_counter_desc.c_str());
     }
 }
 
 void hotspot_partition_calculator::data_analyse()
 {
-    dassert(_partition_stat_histories.back().size() == _hot_points.size(),
-            "partition counts error, please check");
-    std::vector<double> data_samples;
-    data_samples.reserve(_partition_stat_histories.size() * _hot_points.size());
-    auto temp_data = _partition_stat_histories;
-    double table_qps_sum = 0, standard_deviation = 0, table_qps_avg = 0;
-    int sample_count = 0;
-    while (!temp_data.empty()) {
-        for (const auto &partition_data : temp_data.front()) {
-            if (partition_data.total_qps - 1.00 > 0) {
-                data_samples.push_back(partition_data.total_qps);
-                table_qps_sum += partition_data.total_qps;
-                sample_count++;
+    dcheck_eq(_partitions_stat_histories.back().size(), _hot_points.size());
```

Comment:
Are `_partitions_stat_histories.back().size()` and `_hot_points.size()` both expected to be equal to app partition_count? Besides add some log for this assert.